pkg/controller/plan/controller.go

```diff
@@ -257,6 +257,9 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 	// The plan is updated as needed to reflect status.
 	result.RequeueAfter, err = r.execute(plan)
 	if err != nil {
+		if err = r.updatePlanStatus(plan); err != nil {
```

Comment:
```suggestion
		if updateErr = r.updatePlanStatus(plan); updateErr != nil {
```

in this case we don't actuall change the original err, this is good, we do want to return the `execute` error
but reading this code is confusing, a user may think we change the err, and will return the `updatePlanStatus` error

to make it easier to read i will chose a different var name for the `updatePlanStatus` error

a. It's immediately obvious that you're capturing a different error for the update operation
b. No risk of accidentally overwriting the original err in some future code update

Reply:
Thank you for noticing, fixing suggestion, fiding `=` to` :=`

`if updateErr := r.updatePlanStatus(plan); updateErr != nil {`